models/domains_tlds.php

```diff
@@ -432,6 +438,155 @@ public function edit($tld, array $vars)
         }
     }
 
+    /**
+     * Validates if a package will require a module migration.
+     *
+     * @param string $tld The TLD to validate
+     * @param int $new_module_id The ID of the new module
+     * @return bool True if the package needs to be migrated to the new module, false otherwise
+     */
+    private function requiresModuleMigration($tld, $new_module_id)
+    {
+        Loader::loadComponents($this, ['Record']);
+
+        // Check if there is any existing package using the new module
+        $package = $this->getTldPackageByModuleId($tld, $new_module_id);
+
+        if (!empty($package)) {
+            return true;
+        }
+
+        // Get package id
+        $tld = $this->get($tld);
+        $package_id = isset($tld->package_id) ? $tld->package_id : null;
+
+        if (empty($package_id)) {
+            return false;
+        }
+
+        // Check if the package is using the same module id
+        $package = $this->Record->select()
+            ->from('packages')
+            ->where('id', '=', $package_id)
+            ->where('company_id', '=', Configure::get('Blesta.company_id'))
```

Comment:
Using Configure::get('Blesta.company_id') makes the module rather unfriendly to the API and Plugins.  Let's add $company_id = null or $filters = [] parameters instead so that the program calling the model has more control.  The same applies to other uses in this model.  

You could still default 

```
if ($company_id === null) {
    $company_id = Configure::get('Blesta.company_id');
}
```